aten/src/ATen/native/cuda/VolumetricChannelwiseConvolution.cu

```diff
@@ -0,0 +1,845 @@
+// adopted from SpatialDepthwiseConvolution.cu
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/Utils.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/CUDAContext.h>
+
+#include <THC/THCReduceApplyUtils.cuh>
+#include <THCUNN/SharedMem.cuh>
+
+
+#include <THC/THCDeviceTensor.cuh>
+#include <THC/THCDeviceTensorUtils.cuh>
+#include <THC/THCNumerics.cuh>
+#include <THC/THCSortUtils.cuh>
+#include <THC/THCTensorMathReduce.cuh>
+#include <THCUNN/common.h>
+#include <algorithm>
+
+namespace at {
+namespace native {
+
+namespace {
+
+const int WARP_SIZE = 32;
+// Crude benchmarks suggest 256 is better than 512 and 1024
+// TODO: Autotune/use better heuristics, improve speed more.
+const int MAX_BLOCK_SIZE = 256;
+
+static int getGradParamsNumThreads(int batchSize){
+//warp per item in a batch, up to a maximum
+   return std::min(batchSize * WARP_SIZE, MAX_BLOCK_SIZE);
+}
+
+// Your regular forward pass hopefully
+template <typename T, typename accT, typename IndexType, int kSize>
+__global__ void depthwiseConv3DOutput(
+    const Tensor& input,
```

Comment:
you cannot send Tensors as arguments to the kernels, send raw data pointers like here https://github.com/pytorch/pytorch/blob/be3808d3b1164a0073f85fc0a58f2c7c27b2ae40/aten/src/ATen/native/cuda/UpSampleNearest2d.cu#L173-L185